src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,94 @@
+package duke.storage;
+
+import duke.tasks.*;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public String getFilePath() {
+        return filePath;
+    }
+
+    public void appendToFile(String filePath, String textToAppend) throws IOException {
+        FileWriter fw = new FileWriter(filePath, true); // create a FileWriter in append mode
+        fw.write(textToAppend);
+        fw.close();
+    }
+
+    public void writeMarkingToFile(String filePath, String mark, int line) throws IOException {
+        ArrayList<String> fileContent = new ArrayList<>(Files.readAllLines(Paths.get(filePath), StandardCharsets.UTF_8));
+        String s = fileContent.get(line);
+        ArrayList<String> taskDetails = new ArrayList<>(Arrays.asList(s.split(" \\| ")));
+        taskDetails.set(1, mark);
+        String task = "";
+        for (int i = 0; i < taskDetails.size() - 1; i++) {
+            task += taskDetails.get(i) + " | ";
+        }
+        task += taskDetails.get(taskDetails.size() - 1);
+        fileContent.set(line, task);
+        Files.write(Paths.get(filePath), fileContent, StandardCharsets.UTF_8);
+    }
+
+    public void deleteTaskInFile(String filePath, int line) throws IOException {
+        ArrayList<String> fileContent = new ArrayList<>(Files.readAllLines(Paths.get(filePath), StandardCharsets.UTF_8));
+        fileContent.remove(line);
+        Files.write(Paths.get(filePath), fileContent, StandardCharsets.UTF_8);
+    }
+
+    public ArrayList<Task>readTaskList() throws IOException {
```

Comment:
Very minor nitpick but consider adding a space between ArrayList<Task> and readTaskList to adhere to the coding standard
```suggestion
    public ArrayList<Task> readTaskList() throws IOException {
```